File: SampleCode/RestApi/Clients/Java/src/main/java/io/swagger/client/model/UnratedItem.java
/*
 * Emby Server REST API
 * 
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.annotations.SerializedName;
import java.io.IOException;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

/**
 * Gets or Sets UnratedItem
 */
@JsonAdapter(UnratedItem.Adapter.class)
public enum UnratedItem {
  MOVIE("Movie"),
  TRAILER("Trailer"),
  SERIES("Series"),
  MUSIC("Music"),
  GAME("Game"),
  BOOK("Book"),
  LIVETVCHANNEL("LiveTvChannel"),
  LIVETVPROGRAM("LiveTvProgram"),
  CHANNELCONTENT("ChannelContent"),
  OTHER("Other");

  private String value;

  UnratedItem(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static UnratedItem fromValue(String input) {
    for (UnratedItem b : UnratedItem.values()) {
      if (b.value.equals(input)) {
        return b;
      }
    }
    return null;
  }

  public static class Adapter extends TypeAdapter<UnratedItem> {
    @Override
    public void write(final JsonWriter jsonWriter, final UnratedItem enumeration) throws IOException {
      jsonWriter.value(String.valueOf(enumeration.getValue()));
    }

    @Override
    public UnratedItem read(final JsonReader jsonReader) throws IOException {
      Object value = jsonReader.nextString();
      return UnratedItem.fromValue((String)(value));
    }
  }
}

File: SampleCode/RestApi/Net.RestSharp/EmbyClient.Dotnet/Model/UnratedItem.cs
/*
 * EmbyClient.Dotnet
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SwaggerDateConverter = EmbyClient.Dotnet.Client.SwaggerDateConverter;

namespace EmbyClient.Dotnet.Model
{
    /// <summary>
    /// Defines UnratedItem
    /// </summary>
    [JsonConverter(typeof(StringEnumConverter))]
        public enum UnratedItem
    {
        /// <summary>
        /// Enum Movie for value: Movie
        /// </summary>
        [EnumMember(Value = "Movie")]
        Movie = 1,
        /// <summary>
        /// Enum Trailer for value: Trailer
        /// </summary>
        [EnumMember(Value = "Trailer")]
        Trailer = 2,
        /// <summary>
        /// Enum Series for value: Series
        /// </summary>
        [EnumMember(Value = "Series")]
        Series = 3,
        /// <summary>
        /// Enum Music for value: Music
        /// </summary>
        [EnumMember(Value = "Music")]
        Music = 4,
        /// <summary>
        /// Enum Game for value: Game
        /// </summary>
        [EnumMember(Value = "Game")]
        Game = 5,
        /// <summary>
        /// Enum Book for value: Book
        /// </summary>
        [EnumMember(Value = "Book")]
        Book = 6,
        /// <summary>
        /// Enum LiveTvChannel for value: LiveTvChannel
        /// </summary>
        [EnumMember(Value = "LiveTvChannel")]
        LiveTvChannel = 7,
        /// <summary>
        /// Enum LiveTvProgram for value: LiveTvProgram
        /// </summary>
        [EnumMember(Value = "LiveTvProgram")]
        LiveTvProgram = 8,
        /// <summary>
        /// Enum ChannelContent for value: ChannelContent
        /// </summary>
        [EnumMember(Value = "ChannelContent")]
        ChannelContent = 9,
        /// <summary>
        /// Enum Other for value: Other
        /// </summary>
        [EnumMember(Value = "Other")]
        Other = 10    }
}
